### PR TITLE
Refactor: changed legacy way to update database for Hero version integrate

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Operations used instead.

## Testing notes:
1. I am actually not sure how to test all use cases (with representations, unarchiving them etc), so any steps to test this properly are welcomed
2. Publish any family that produces Hero version (I have tested in on `model`).

### CODE REVIEW

**Comment by BigRoy** [comment]
Might be best to update the title of this PR to state that it's only about Hero Versions?

File: openpype/client/operations.py [hunk 1]
```diff
@@ -162,6 +163,34 @@ def new_version_doc(version, subset_id, data=None, entity_id=None):
     }
 
 
+def new_hero_version_doc(version_id, parent_id, data=None, entity_id=None):
```
Comment (iLLiCiTiT) [inline openpype/client/operations.py:166]:
```suggestion
def new_hero_version_doc(version_id, subset_id, data=None, entity_id=None):
```

Comment (iLLiCiTiT) [inline openpype/client/operations.py:166]:
```suggestion
def new_hero_version_doc(version_id, subset_id, data=None, entity_id=None):
```

File: openpype/plugins/publish/integrate_hero_version.py [hunk 2]
```diff
@@ -187,35 +191,29 @@ def integrate_instance(
...
-        }
-        schema.validate(new_hero_version)
+        op_session = OperationsSession()
 
-        # Don't make changes in database until everything is O.K.
-        bulk_writes = []
+        new_hero_version = new_hero_version_doc(
+            src_version_entity["_id"],
+            src_version_entity["parent"]
+        )
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_hero_version.py:199]: Rather pass the id to the `new_hero_version_doc` then change it afterwards
```suggestion
        entity_id = None
        if old_version:
            entity_id = old_version["_id"]
        new_hero_version = new_hero_version_doc(
            src_version_entity["_id"],
            src_version_entity["parent"],
            entity_id=entity_id
        )
```

File: openpype/plugins/publish/integrate_hero_version.py [hunk 3]
```diff
@@ -382,34 +380,37 @@ def integrate_instance(
...
+                        update_data
                     )
 
                 # Create representation
                 else:
-                    repre["_id"] = ObjectId()
-                    bulk_writes.append(
-                        InsertOne(repre)
-                    )
+                    repre["_id"] = _create_or_convert_to_mongo_id(None)
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_hero_version.py:411]: Why?

File: openpype/plugins/publish/integrate_hero_version.py [hunk 4]
```diff
@@ -430,28 +431,22 @@ def integrate_instance(
...
+                               "_id": archived_repre["_id"],
+                               "type": archived_repre["type"]}
+                    op_session.update_entity(project_name,
+                                             archived_repre["type"],
+                                             archived_repre["_id"],
+                                             changes)
                 else:
                     repre["old_id"] = repre["_id"]
-                    repre["_id"] = ObjectId()
+                    repre["_id"] = _create_or_convert_to_mongo_id(None)
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_hero_version.py:444]:
```suggestion
                    repre.pop("_id")
```

Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_hero_version.py:444]:
```suggestion
                    repre["old_id"] = repre.pop("_id")
```

File: openpype/plugins/publish/integrate_hero_version.py [hunk 5]
```diff
@@ -187,35 +190,33 @@ def integrate_instance(
...
+        )
 
         if old_version:
             self.log.debug("Replacing old hero version.")
-            bulk_writes.append(
-                ReplaceOne(
-                    {"_id": new_hero_version["_id"]},
-                    new_hero_version
-                )
+            new_hero_version["_id"] = old_version["_id"]
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_hero_version.py:206]: This line should not be needed anymore.
```suggestion
```

**Review by iLLiCiTiT — APPROVED** [review]
One small suggestion but otherwise it works.

**Pull request merged**